### ISSUE TITLE
Getting the new comic makes an excessive amount of requests?

### ISSUE THREAD

**Issue by mckennajones** [Blastitt/DailyXKCD#7]
So I was looking at my PiHole admin page today and I noticed that I had over 12000 dns requests to XKCD.com

These all occurred within a few minutes around the time it was trying to fetch the new comic. My refresh interval is set to the default value.

Any ideas why this might be happening?

**Comment by Blastitt** [comment]
The only times it should make a request are:

1. When the application starts up.
2. When the update interval you set in the config is reached
3. When either of the above happen on a Tuesday, Thursday, Saturday, or
Sunday.

Are you sure those 12000 DNS requests occurred within a few minutes and
that your update interval is set correctly? Remember the value for the
update interval is in milliseconds, not minutes or hours. If this is all
correct, then maybe the application is crashing and starting multiple times?

How often does the visibly change? If it really is requesting that much, on
one of the above stated days it should change rapidly, as each request is
for a random comic.

**Comment by mckennajones** [comment]
Below you can see that the spikes in requests occurred 10 hours apart, because I haven't changed the default update interval. 
![xkcd](https://cloud.githubusercontent.com/assets/7456446/23244966/c23d548c-f93d-11e6-84c6-4eef1a26e441.PNG)

Here's a screenshot of my MM config, just to be sure.
![mmconfig](https://cloud.githubusercontent.com/assets/7456446/23244959/b47804b4-f93d-11e6-96e6-6772f4cdcf26.PNG)

**Comment by Blastitt** [comment]
Can you check your mirror logs and see how often it says:

"XKCD: Getting comic."

for one of those spikes in DNS requests?  This will help me determine
whether it's the front end or back end script that contains the error.

**Comment by mckennajones** [comment]
Interestingly I don't see that in my logs, this could be because I just restarted MM?

However, I do see `Notification: GET_COMIC Payload: [object Object]` often.

`cat /home/pi/.pm2/logs/mm-out-0.log | grep GET_COMIC | wc -l` outputs 60355

**Comment by Blastitt** [comment]
I believe I corrected the issue and it should be fixed in the latest build. The problem was that the timer for getting a new comic was set every time a new comic was returned. It turns out each timer automatically resets once it executes the callback function, so this lead to many staggered update timers going off around the same time. I removed the code that created new timers and set a single timer on startup.

Thank you for bringing this issue to my attention, as I hadn't noticed my Mirror was also doing this until you said something. Bug reporters and users like you are the reason I like making open source software.
